Collect and report statistics when parsing paired reads

m-8k opened the pull request:
- report discarded reads/pairs via warning
- simplifies debugging "no valid reads" error

m-8k committed:
collect statistics when parsing paired reads

- report discarded reads/pairs via warning
- simplifies debugging error (ErrorCode::no_valid_reads)

kpj commented on lib/statistics_to_messages.cpp:
```diff
@@ -0,0 +1,88 @@
+    return to_percentage(x, statistics.pairs.total_count);
+  };
+
+  if (statistics.reads.unsupported_attribute) {
```
Would it be possible to list here which used attribute was unsupported?

m-8k replied:
Thanks for the suggestion! Since the attribute could be different for every read, I only see the option of splitting up the warning into individual warnings per attribute/bit so far. Is this what you have in mind?

kpj replied:
Yes, that would be useful. But if that is too tedious to implement, it is also great like this.

m-8k replied:
What do you think about the following suggestion: We could keep the current general warning for now, and add specific warnings as we encounter them? I think this would have the advantage of being able to show more precise warnings for combinations of FLAGs that are common. (I'm concerned that a warning about a single one of these attribute is difficult to interpret, too, and requires inspection of the SAM file anyway. Is this concern valid?)

kpj replied:
Yes, let's keep it like this for now.

Not sure how precise we can actually make these warnings. Wouldn't we always just list what was not understood?

m-8k replied:
Perfect, thanks! Yes, that's possible; let's decide when we encounter a use case.

kpj commented on lib/statistics_to_messages.cpp:
```diff
@@ -0,0 +1,88 @@
+  if (statistics.pairs.align_score_fraction_too_low) {
+    r.push_back(
+        to_pairs_percentage(statistics.pairs.align_score_fraction_too_low) +
+        " of the read pairs were discarded because the alignment score "
```
Could we mention here which tag this alignment score is stored in?

m-8k replied:
Thanks, good idea! The SAM TAG that the alignment score fraction is based on ("AS") is now mentioned in the message.

kpj replied:
Great!

m-8k committed:
fix typo in error message

m-8k committed:
add information about alignment score fraction

- alignment score fraction is value of AS divided by length of sequence
- see functions parseSAMpaired, parse_sam_line
